vars/publishToArtifactsProdBucket.groovy

```diff
@@ -19,18 +19,26 @@
 void call(Map args = [:]) {
     lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
     println('Signing the artifacts')
-    signArtifacts(
+    if (args.signingPlatform == 'windows' || args.signingPlatform == 'mac') {
+        signArtifacts(
+            artifactPath: args.source,
+            platform: args.signingPlatform,
+            overwrite: args.sigOverwrite ?: true
+            )
+    } else {
+        signArtifacts(
```

Comment:
This change will just sign `windows` zip with windows signing, which should not be and should just use the normal detached signature method.

Thoughts?

Reply:
That is what we want. Windows does not do detached signature. It signs actual binary. Only pgp gives detached signature

Reply:
I am talking about the windows zip we release alongside linux tar.
https://artifacts.opensearch.org/releases/bundle/opensearch/2.8.0/opensearch-2.8.0-windows-x64.zip
https://artifacts.opensearch.org/releases/bundle/opensearch/2.8.0/opensearch-2.8.0-windows-x64.zip.sig

Reply:
That uses PGP signing not windows signing.

Reply:
Yes that is my point your change will just make the zip not able to be signed with detached sig, aka pgp.

Reply:
Oh this is not related to artifact promotion right? Only client release?

Reply:
How? The library you are looking at is `vars/publishToArtifactsProdBucket.groovy` I did not change anything in `signArtiacts` library.
Also for above artifacts you mentioned. args.signingPlatform == linux

Reply:
Yes you are right! We can reuse this library but for now it is not affecting any promoteArifacts workflow

Reply:
Got it, thanks.